File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>

string cipher(string, int);

int main(int argc, string argv[])
{
    string text, ciphered;
    int shift;

    // Check arguments.
    if (argc != 2 || (shift = atoi(argv[1])) <= 0)
    {
        printf("Usage: ./caesar k,\nwhere k is an integer and it's greater than 0.\n");
        return 1;
    }

    // Request user string.
    text = get_string("plaintext: ");

    // Encipher the text.
    ciphered = cipher(text, shift);

    // Print the result.
    printf("%s\n", ciphered);
}

string cipher(string text, int shift)
{
    int i, c;
    char upper;

    // Chipher the text.
    for (i = 0; text[i] != '\0'; i++)
    {
        // Character must be alphabetic.
        if (!isalpha(text[i])) continue;

        upper = toupper(text[i]);
        c = (upper - 'A' + shift) % 26;
        if (upper == text[i])
            text[i] = 'A' + c;
        else
            text[i] = 'a' + c;
    }
    return text;
}

File: pset6/sentimental/caesar/caesar.py
# Implement Caesar cipher

import sys
from cs50 import get_string

def main():
    # Check command-line argument
    if len(sys.argv) != 2:
        print("Usage: python caesar key")
        sys.exit(1)
    key = int(sys.argv[1])

    plaintext = get_string("What to cipher? ")

    enciphered = encipher_caesar(plaintext, key)

    # Print the result
    print(f"ciphertext: {enciphered}")

# Encipher text with Caesar’s algorithm
def encipher_caesar(text, key):
    enciphered = list(text)

    # Encipher every character in the text
    for i in range(len(enciphered)):

        # Only for alphabetic characters
        char = enciphered[i]
        if char.isalpha():

            # Store offset of case
            offset = 65 if char.isupper() else 97

            shift = (ord(char) - offset + key) % 26
            enciphered[i] = chr(offset + shift)
    return ''.join(enciphered)


if __name__ == "__main__":
    main()